### PR TITLE
Provide an option to limit the maximum concurrent request to consul.

### PR DESCRIPTION
**Pull request by MaruHyl**
Recommited.
Close https://github.com/prometheus/consul_exporter/issues/102.
Close https://github.com/prometheus/consul_exporter/pull/94.

### CODE REVIEW

File: consul_exporter.go [hunk 1]
```diff
@@ -448,6 +464,7 @@ func main() {
 	kingpin.Flag("consul.server-name", "When provided, this overrides the hostname for the TLS certificate. It can be used to ensure that the certificate name matches the hostname we declare.").Default("").StringVar(&opts.serverName)
 	kingpin.Flag("consul.timeout", "Timeout on HTTP requests to the Consul API.").Default("500ms").DurationVar(&opts.timeout)
 	kingpin.Flag("consul.insecure", "Disable TLS host verification.").Default("false").BoolVar(&opts.insecure)
+	kingpin.Flag("consul.request-limit", "Limit the maximum number of concurrent requests to consul.").Default("0").IntVar(&opts.requestLimit)
```
[the remark this answers is not in the document]
Comment (MaruHyl) [reply consul_exporter.go]: Done

File: consul_exporter.go [hunk 2]
```diff
@@ -448,6 +464,7 @@ func main() {
 	kingpin.Flag("consul.server-name", "When provided, this overrides the hostname for the TLS certificate. It can be used to ensure that the certificate name matches the hostname we declare.").Default("").StringVar(&opts.serverName)
 	kingpin.Flag("consul.timeout", "Timeout on HTTP requests to the Consul API.").Default("500ms").DurationVar(&opts.timeout)
 	kingpin.Flag("consul.insecure", "Disable TLS host verification.").Default("false").BoolVar(&opts.insecure)
+	kingpin.Flag("consul.request-limit", "Limit the maximum number of concurrent requests to Consul.").Default("0").IntVar(&opts.requestLimit)
```
Comment (roidelapluie) [inline consul_exporter.go:467]: this only covers the services api

Comment (roidelapluie) [reply consul_exporter.go:467]: maybe specify that 0 = no limit

Comment (simonpasquier) [reply consul_exporter.go:467]: > this only covers the services api

This is the only API endpoint that is called concurrently for the time being. If other endpoints were concerned in the future, I'd expect the same limit to be applied.

Comment (roidelapluie) [reply consul_exporter.go:467]: good point

Comment (MaruHyl) [reply consul_exporter.go]: done

**Comment by simonpasquier** [comment]
Thanks!

**Pull request merged**